src/lib/app-provider/hooks/useAttachFunds.ts

```diff
@@ -0,0 +1,43 @@
+import type { Coin } from "@cosmjs/stargate";
+import { useCallback } from "react";
+
+import { AttachFundsType } from "lib/components/fund/types";
+import { fabricateFunds, sortDenoms } from "lib/utils";
+
+import { useChainRecordAsset } from "./useChainRecordAsset";
+
+export const useAttachFunds = () => {
+  const getAssetInfo = useChainRecordAsset();
+
+  return useCallback(
+    (
+      attachFundsOption: AttachFundsType,
+      assetsJsonStr: string,
+      assetsSelect: Coin[]
+    ) => {
+      const assetsSelectWithPrecision = assetsSelect.map((coin) => {
+        return {
+          ...coin,
+          precision: getAssetInfo(coin.denom)?.precision ?? 1,
```

Comment:
let the argument type be `Option<number>`, then let `exponentify()` handle the default case